### PR TITLE
Fix error risk pingback requests

### PR DESCRIPTION
**Pull request by dzungtran**

### CODE REVIEW

File: src/app/code/community/Paymentwall/Paymentwall/Model/Method/Abstract.php [hunk 1]
```diff
@@ -130,15 +130,24 @@ private function prepareDeliveryData(Mage_Sales_Model_Order $order, $ref) {
...
         if ($order) {
-            $invoice = $order->prepareInvoice()
-                ->setTransactionId($pingback->getReferenceId())
-                ->addComment("Invoice created by Paymentwall")
-                ->register();
 
-            $this->payInvoice($pingback, $invoice);
+            $payment = $order->getPayment();
+            $payment->setTransactionId($pingback->getReferenceId())
+                ->setPreparedMessage('Invoice created by PaymentWall module')
```
Comment (sevastyanovio) [inline src/app/code/community/Paymentwall/Paymentwall/Model/Method/Abstract.php]: via Paymentwall

Comment (dzungtran) [inline src/app/code/community/Paymentwall/Paymentwall/Model/Method/Abstract.php]: updated

File: src/app/code/community/Paymentwall/Paymentwall/Model/Method/Pwbrick.php [hunk 2]
```diff
@@ -139,9 +140,23 @@ public function submitRecurringProfile(Mage_Payment_Model_Recurring_Profile $pro
...
+
         $subscriptionData = $this->prepareSubscriptionData($profile, $quote);
         $paymentwallSubscription = new Paymentwall_Subscription();
-        $paymentwallSubscription->create($subscriptionData);
+        $paymentwallSubscription->create(array_merge(
+            $this->prepareSubscriptionData($profile, $quote),
+            $this->prepareCardInfo($payment),
+            array(
+                'custom[integration_module]' => 'magento',
+                'uid' => $customerId // for Pingback Request
```
Comment (sevastyanovio) [inline src/app/code/community/Paymentwall/Paymentwall/Model/Method/Pwbrick.php]: DRY

File: src/app/code/community/Paymentwall/Paymentwall/Model/Method/Pwbrick.php [hunk 3]
```diff
@@ -85,6 +85,7 @@ public function capture(Varien_Object $payment, $amount) {
             $this->prepareUserProfile($payment->getOrder()), // for User Profile API
             $this->prepareCardInfo($payment),
             array(
+                'custom[integration_module]' => 'magento',
                 'uid' => $customerId // for Pingback Request
```
Comment (davex69) [inline src/app/code/community/Paymentwall/Paymentwall/Model/Method/Pwbrick.php]: DRY

File: src/app/code/community/Paymentwall/Paymentwall/Model/Method/Pwbrick.php [hunk 4]
```diff
@@ -85,6 +85,7 @@ public function capture(Varien_Object $payment, $amount) {
             $this->prepareUserProfile($payment->getOrder()), // for User Profile API
             $this->prepareCardInfo($payment),
             array(
+                'custom[integration_module]' => 'magento',
```
Comment (sevastyanovio) [inline src/app/code/community/Paymentwall/Paymentwall/Model/Method/Pwbrick.php]: DRY

File: src/app/code/community/Paymentwall/Paymentwall/Model/Pingback.php [hunk 5]
```diff
@@ -77,6 +77,10 @@ protected function processPingbackOrder(Paymentwall_Pingback $pingback) {
             try {
                 if ($pingback->isDeliverable()) {
 
+                    if($order->getState() == Mage_Sales_Model_Order::STATE_PROCESSING){
```
Comment (sevastyanovio) [inline src/app/code/community/Paymentwall/Paymentwall/Model/Pingback.php:80]: spaces

File: src/app/code/community/Paymentwall/Paymentwall/Model/Method/Pwbrick.php [hunk 6]
```diff
@@ -139,9 +140,23 @@ public function submitRecurringProfile(Mage_Payment_Model_Recurring_Profile $pro
         $this->initPaymentwallConfig();
         $quote = Mage::getSingleton('checkout/session')->getQuote();
 
+        $customerId = $_SERVER['REMOTE_ADDR'];
```
Comment (sevastyanovio) [inline src/app/code/community/Paymentwall/Paymentwall/Model/Method/Pwbrick.php]: it's not cusromer id

Comment (dzungtran) [inline src/app/code/community/Paymentwall/Paymentwall/Model/Method/Pwbrick.php]: I can't leave this field empty

File: src/app/code/community/Paymentwall/Paymentwall/Model/Pingback.php (hunk 5 above)
Comment (dzungtran) [inline src/app/code/community/Paymentwall/Paymentwall/Model/Pingback.php:80]: updated

**Pull request merged**